### PR TITLE
Add child run and run metrics methods

### PR DESCRIPTION
**Pull request by diondrapeck**

### CODE REVIEW

File: R/hyperdrive.R [hunk 1]
```diff
@@ -692,3 +692,85 @@ get_child_run_hyperparameters <- function(hyperdrive_run) {
...
   hyperdrive_run$get_metrics()
 }
+
+#' Create a child run
+#'
+#' @description
+#' Create a child run. This is used to isolate part of a run into a subsection.
+#' @param parent_run The `HyperDriveRun` object.
+#' @param name An optional name for the child run, typically specified for a "part"
+#' @param run_id An optional run ID for the child, otherwise it is auto-generated.
```
Comment (ninghu) [inline R/hyperdrive.R]: can't parent_run be any run?

Comment (ninghu) [reply R/hyperdrive.R]: These new child run apis are not limited to hyperdriver. It should be part of run.R

File: R/hyperdrive.R [hunk 2]
```diff
@@ -692,3 +692,85 @@ get_child_run_hyperparameters <- function(hyperdrive_run) {
...
+#' @return A list of child runs.
+#' @export
+#' @md
+get_child_runs <- function(parent_run,
+                           recursive = FALSE,
+                           tags = NULL,
+                           properties = NULL,
+                           type = NULL,
+                           status = NULL,
+                           rehydrate_runs = TRUE) {
```
Comment (ninghu) [inline R/hyperdrive.R]: I don't see this parameter in the spec. Did I miss something?

File: tests/testthat/test_hyperdrive.R [hunk 3]
```diff
@@ -43,19 +43,31 @@ test_that("create hyperdrive config, launch runs, get run metrics", {
...
   
+  # create new child runs
+  extra_child <- create_child_run(hyperdrive_run, run_id = "my_new_child")
+  expect_equal(extra_child$id, "my_new_child")
+
+  extra_children <- create_child_runs(hyperdrive_run, count = 3L)
+  expect_equal(length(extra_children), 3)
+
+  # extra child run
+  child_run <- submit_child_run(hyperdrive_run, hyperdrive_config)
```
Comment (ninghu) [inline tests/testthat/test_hyperdrive.R]: I don't think using submit_child_run in hyperdriver scenario is right. 
https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-manage-runs#submit-child-runs

File: vignettes/hyperparameter-tune-with-keras/hyperparameter-tune-with-keras.Rmd [hunk 4]
```diff
@@ -210,6 +210,8 @@ Wait until hyperparameter tuning is complete before you run more code.
 wait_for_run_completion(hyperdrive_run, show_output = TRUE)
 ```
 
+More child runs can be added later by using `create_child_run()` or `create_child_runs()` to return a Run object or list of Run objects. Alternatively, you can pass a HyperDriveConfig object along with the parent run to `submit_child_run()`.
```
Comment (ninghu) [inline vignettes/hyperparameter-tune-with-keras/hyperparameter-tune-with-keras.Rmd]: same here. My understanding is that create_child_run is for interactive logging scenario.

File: tests/testthat/test_hyperdrive.R [hunk 5]
```diff
@@ -43,19 +43,31 @@ test_that("create hyperdrive config, launch runs, get run metrics", {
...
   expect_equal(length(child_run_metrics), 4)
   
   # find best-performing run
   best_run <- get_best_run_by_primary_metric(hyperdrive_run)
-  
+  expected_best_run <- toString(sorted_child_runs[[1]][1])
   expect_equal(expected_best_run, best_run$id)
   
+  # create new child runs
+  extra_child <- create_child_run(hyperdrive_run, run_id = "my_new_child")
```
Comment (ninghu) [inline tests/testthat/test_hyperdrive.R]: the tests should be moved to test_experiment.R

File: R/hyperdrive.R [hunk 6]
```diff
@@ -692,3 +692,85 @@ get_child_run_hyperparameters <- function(hyperdrive_run) {
...
+#' Create a child run
+#'
+#' @description
+#' Create a child run. This is used to isolate part of a run into a subsection.
+#' @param parent_run The `HyperDriveRun` object.
+#' @param name An optional name for the child run, typically specified for a "part"
+#' @param run_id An optional run ID for the child, otherwise it is auto-generated.
+#' Typically this parameter is not set.
+#' @param outputs Optional outputs directory to track for the child.
+#' @return The child run.
```
Comment (heemanshusuri) [inline R/hyperdrive.R]: also add the type of child run object

File: R/hyperdrive.R (hunk 1 above)
Comment (diondrapeck) [reply R/hyperdrive.R]: Thanks for pointing this out. My bad.

File: tests/testthat/test_hyperdrive.R (hunk 3 above)
Comment (mx-iao) [reply tests/testthat/test_hyperdrive.R]: yeah, `submit_child_run` should be for generic runs and be added to run.R

File: R/hyperdrive.R (hunk 2 above)
Comment (mx-iao) [reply R/hyperdrive.R]: the parameter in Python is preceded by an underscore `_rehydrate_runs` so I didn't add it to the spec. Let me double check w/ the Python team to see if that's something we should expose.

File: tests/testthat/test_experiment.R [hunk 7]
```diff
@@ -124,4 +124,42 @@ test_that("Create an interactive run, log metrics locally.", {
...
+  
+  env <- r_environment("myenv", cran_packages = c("dplyr", "ggplot2"))
+  
+  est <- estimator(tmp_dir_name,
+                   compute_target = existing_compute$name, 
+                   entry_script = script_name,
+                   environment = env)
+  
+  exp <- experiment(ws, "estimator_run")
+  run <- submit_experiment(exp, est)
```
Comment (ninghu) [inline tests/testthat/test_experiment.R]: we can create a parent run using iterative logging api and it returns immediately. i forgot the name of the function, it is called starting_logging or something.

Comment (diondrapeck) [reply tests/testthat/test_experiment.R]: I found it! Thanks so much!

**Pull request merged**